### DISCUSSION TITLE
Is it possible, and if yes using which components, to build office application pages with traditional master-detail records editing?

### DISCUSSION THREAD

**Discussion by aramgharib** [sqlpage/SQLPage#227] [Q&A]
Hi,
I'm genuinely impressed by the simplicity and rapidity with which one can realise nice and dynamic web pages with SQLpage!
I was wondering if I could use this platform to build, for example, an office automation application with several master and detail tables that the user would need to interact with- i.e. create, retreive, edit and delete data. Here's a typical example of "Master*" fields from the master table and "detailx*" records from detail table(s); "Comm1", "+", and "-" being some command buttons acting on the Master/detail records:

![image](https://github.com/lovasoa/SQLpage/assets/8828815/2dc7fdba-9dd5-430e-aab7-51e3d4827b59)

Thanks in advance for any insight.

-- Aram Gharib

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage!

 You can run any insert or select query you want in SQLPage, so I don't see any reason you couldn't implement your office automation application, provided you can structure your user interface to match SQLPage's built-in components (or you are ready to write your own components). 

About having a master table and associated detail tables, you may be interested in this example: https://github.com/lovasoa/SQLpage/tree/main/examples/master-detail-forms

If you encounter any issue while building your application, don't hesitate to write about it here and to ask questions!
